src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -3090,6 +3092,31 @@ namespace winrt::TerminalApp::implementation
         term.RefreshQuickFixMenu();
     }
 
+    void TerminalPage::_WindowSizeChanged(const IInspectable /*sender*/, const Microsoft::Terminal::Control::WindowSizeChangedEventArgs args)
+    {
+        // Raise if:
+        // - Not in quake mode
+        // - Not in fullscreen
+        // - Only one tab exists
+        // - Only one pane exists
+        // else:
+        // - Reset conpty to its original size back
+        if (!WindowProperties().IsQuakeWindow() && !Fullscreen() &&
+            NumberOfTabs() == 1 && _GetFocusedTabImpl()->GetLeafPaneCount() == 1)
+        {
+            WindowSizeChanged.raise(*this, args);
+        }
+        else if (const auto& control{ _GetActiveControl() })
```

Comment:
Okay this is subtle, but this `WindowSizeChanged` handler might be being called for an _inactive_ tab. So like, (as written) an inactive tab might get a `CSI t`, and then we'll reset the size of the _focused_ tab, not the one that raised the event.

Fortunately, this is easy to handle! We've already set the `sender` of this event to the `TermControl` requesting the resize. So you can uncomment `/*sender*/` above, and instead do:

```suggestion
        else if (const auto& control{ sender.try_as<TermControl>() })
```